### PR TITLE
Add Mingw/Msys support

### PR DESCRIPTION
**Pull request by rasa**
Per https://en.wikipedia.org/wiki/Uname , `uname -s` on [Cygwin](https://cygwin.com/), [MinGW](http://www.mingw.org/), [MinGW-w64](https://sourceforge.net/projects/mingw-w64/), [Msys](http://www.mingw.org/wiki/MSYS), [Msys2](http://www.msys2.org/), [git-bash](https://gitforwindows.org/), etc., return the following values:
````
CYGWIN_NT-5.1
CYGWIN_NT-6.1
CYGWIN_NT-6.1-WOW64
CYGWIN_NT-10.0
MINGW32_NT-6.1
MINGW64_NT-6.1
MSYS_NT-6.1
````
Some people [see](https://github.com/ninenines/erlang.mk/issues/632) `MINGW32_NT-6.2` and `MINGW64_NT-6.2` as well.

This patch replaces `uname -s` with `uname -o` (on these systems only). `uname -o` returns more sane values (again, on these systems only):
```
Cygwin # on all versions of Cygwin
Msys   # on all versions of MinGW32, MinGW64, Msys, Msys2, git-bash, etc.
```
Additionally, `USE_CYGPATH` is set for all Msys variants, as it applies to them too (though I have symlinks working fine in Windows 7).

### CODE REVIEW

**Comment by TheLocehiliosan** [comment]
Thanks for the contribution. I can see how the variety of values from cygwin would make using alternates more painful. I'll add some test cases for this when I merge it.

**Comment by rasa** [comment]
Please consider removing the [OS check]( https://github.com/TheLocehiliosan/yadm/blob/master/yadm#L172) for the `yadm.cygwin-copy` option, as I can imagine use cases where the user is not on Windows, but the file system doesn't support symlinks for some [reason](https://cygwin.com/cygwin-ug-net/using.html#pathnames-symlinks).

File: yadm [hunk 1]
```diff
@@ -169,7 +169,7 @@ function alt() {
 
   #; decide if a copy should be done instead of a symbolic link
   local do_copy=0
-  if [[ $OPERATING_SYSTEM == CYGWIN* ]] ; then
+  if [[ $OPERATING_SYSTEM =~ ^(CYGWIN|MSYS) ]] ; then
```
Comment (TheLocehiliosan) [inline yadm]: @rasa I was just merging this into the develop branch, and I noticed something different between your change and your comments in this PR.

You're mention that `uname -o` will be either "Cygwin" or "Msys" in these cases, but when testing the $OPERATING_SYSTEM, you're testing agains the regex `^(CYGWIN|MSYS)`.

Can you see if these should be initial caps or all caps? I do not have a Windows system to test with. My guess is that this should be `^(Cygwin|Msys)` based on the other information, including Wikipedia.

Maybe this doesn't matter too much, as you've suggested universal supporting an option to for alternate "copies". I think I may do just that.

Comment (rasa) [reply yadm]: You are correct:
```
ross@indra ~
$ uname -o
Cygwin
ross@indra ~
$ uname -s
CYGWIN_NT-10.0
```
```
ross@indra:~$ uname -o
Msys
ross@indra:~$ uname -s
MSYS_NT-10.0
```

The commit https://github.com/TheLocehiliosan/yadm/pull/102/commits/54f7cbcebe835b098a23998a16eb11f4d643eeb1 is wrong. https://github.com/TheLocehiliosan/yadm/pull/102/commits/315ad0873e843b1cfa7f2a9933e15dde798cba24 is correct. Sorry for the mixup.

Comment (TheLocehiliosan) [reply yadm]: No worries. Your comments were enough for me to realize it was probably like that.

I did actually strip out the OS restriction on that feature (changing the option to `yadm.alt-copy`). But knowing the actual values for OPERATING_SYSTEM will be helpful for documentation examples, etc.  Thanks.

**Pull request merged**